internal/node/pubsub/pubsub.go

```diff
@@ -10,14 +10,14 @@ type Event interface {
 }
 
 type Publisher[E Event] interface {
-	PublishEvent(*E) error
-	GetChan() <-chan *E
+	PublishEvent(E) error
```

Comment:
what motivated this change?

Reply:
If the event is an interface, this implies passing a pointer to an interface, which is weird.